solanaj/src/main/java/org/p2p/solanaj/utils/crypto/Pbkdf2HashGenerator.kt

```diff
@@ -0,0 +1,56 @@
+package org.p2p.solanaj.utils.crypto
+
+import java.security.SecureRandom
+import javax.crypto.SecretKeyFactory
+import javax.crypto.spec.PBEKeySpec
+
+private const val SALT_GENERATE_ALGORITHM = "SHA1PRNG"
+private const val SALT_SIZE = 16
+
+private const val PIN_HASH_ALGORITHM = "PBKDF2WithHmacSHA1"
+
+private const val PBKDF2_ITERATIONS = 5000
+private const val PBKDF2_DEFAULT_LENGTH_BYTES = 64
+
+class Pbkdf2Hash(
+    val hashResultInHex: String,
+    val hashSalt: ByteArray
+)
+
+class Pbkdf2HashGenerator {
```

Comment:
same logic as in HashingUtils

except now we can mock this hashing function and see what's its doing inside